core/cas-server-core-events-api/src/main/java/org/apereo/cas/support/events/listener/CasAuthenticationAuthenticationEventListener.java

```diff
@@ -110,4 +120,18 @@ public void handleCasRiskyAuthenticationDetectedEvent(final CasRiskyAuthenticati
         dto.setPrincipalId(event.getAuthentication().getPrincipal().getId());
         this.casEventRepository.save(dto);
     }
+
+    @Override
+    public void handleCasRequestSingleLogoutEvent(final CasRequestSingleLogoutEvent event) throws Throwable {
+        try {
+            val ticket = event.getTicketGrantingTicket();
+            LOGGER.debug("Performing single logout for expired ticket-granting ticket [{}]", ticket.getId());
+            val request = SingleLogoutExecutionRequest.builder()
+                .ticketGrantingTicket(ticket)
+                .build();
+            logoutManager.performLogout(request);
```

Comment:
Please log all failed attempts, if any.

Reply:
I catch `Throwable` and use `LoggingUtils.error(LOGGER, e);`: isn't it enough?

Reply:
Almost; you'll need to also examine the results returned via `performLogout` and log what is failed.

Reply:
I'm not sure what you want me to examine. It returns the list of the back-channel processed + front-channel not processed logout requests. Currently, in the `DefaultTicketRegistryCleaner`, it is ignored. I have the feeling we should do the same here.

Reply:
Interesting! I didn't realize it is also ignored there. We can keep it the same here, but I think this should be changed in all areas to log out anything that fails. Something like this perhaps:

```java
                    val results = logoutManager.performLogout(request);
                    results.stream().filter(r -> r.getStatus() == LogoutRequestStatus.FAILURE)
                        .forEach(r -> LOGGER.warn("Logout request for [{}] has failed", r.getTicketId()));
```

If you feel like it, I think both places should be changed. If not, we can do that later. Does that make sense?